File: LeetCode/python3/70.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1 or n == 0:
            return 1
        prev, curr = 1, 1
        for i in range(2, n + 1):
            temp = curr
            curr += prev
            prev = temp
        return curr

File: LeetCode/cpp/70.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n < 2) return 1;
        if (n == 2) return 2;
        int prev = 1, last = 2, sum;
        for (int i = 3; i <= n; ++i) {
            sum = prev + last;
            prev = last;
            last = sum;
        }
        return sum;
    }
};